digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataImporter/PointAssetCsvImporter.scala

```diff
@@ -112,6 +147,24 @@ trait PointAssetCsvImporter extends CsvDataImporterOperations {
     }
   }
 
+  def csvLaneValidator(optLaneType: Option[Int], optLaneNumber: Option[String]): (Boolean, List[String]) = {
+    (optLaneType, optLaneNumber) match {
+      case (_, Some(lane)) if lane.trim.nonEmpty && !lane.matches("^([1-3][1-9])$") =>
+        (false, List("Invalid lane"))
+      case (Some(laneType), Some(laneNumber)) if laneType != LaneType.Unknown.value && laneNumber.trim.nonEmpty =>
+        val isMainTypeAndWrongLaneNumber = laneType == LaneType.Main.value && !LaneNumber.isMainLane(laneNumber.toInt)
+        val isNotMainTypeAndIsMainLaneNumber = laneType != LaneType.Main.value && LaneNumber.isMainLane(laneNumber.toInt)
```

Comment:
One is not the opposite of the other?
Cannot be:
```Scala
val isNotMainTypeAndIsMainLaneNumber = !isMainTypeAndWrongLaneNumber

Reply:
One is not the opposite of the other. For example for a LaneType.Main and LaneNumber = 11 both of them will return false. If you invert the result there will be an error on a successful case